### PR TITLE
2.1.0

### LINKED ISSUE #815: Network in modal not synchronized with connected wallet's network
**Describe the bug**
The network (shown in the modal) does not seem to be synced up with Metamask (or other wallet UI) after page reloads or internal routing changes

**SDK Version**
 - Version: 2.0.0-rc.2 + whatever is currently on the try-it-out

**To Reproduce**
1. Go to https://web3modal.com/try-it-out
2. Connect a Metamask/Brave wallet (maybe others too)
3. Change the network to anything but Ethereum (e.g. Polygon)
4. Reload the page, or trigger an internal route change (e.g. click on "Introduction" in the left nav)
6. Navigate back to the try it out page, see it showing Polygon balance but with network as Ethereum

**Expected behavior**
The network on the modal should always be synchronized with the wallet selected in the Metamask/other wallet UI.

**Screenshots**
Screenshot 2023-01-09 at 4 46 14 PM

**Desktop (please complete the following information):**
 - OS: Mac (Ventura 13.1)
 - Browser: tested Firefox (108.0.2) and Brave (1.46.153)

**Additional context**

### LINKED ISSUE #824: Allow to skip rpc calls for balance / ens
**Describe the bug**
As soon as a wallet is connected through the web3modal. A bunch of RPC requests are being made. From what I can see, all the calls are to the following contracts which belong to ENS.

- 0x4976fb03c32e5b8cfe2b6ccb31c09ba78ebaba41
- 0xa2c122be93b0074270ebee7f6b7292c7deb45047
- 0x00000000000c2e074ec69a0dfb2997ba6c7d2e1e

I am not calling any of the ENS methods (e.g. `fetchEnsName` or `fetchEnsAvatar`).

Also, one of the calls is for `eth_getBalance`. Again, I'm not calling the `fetchBalance` method.

Are these calls intended? If they are, how can I access their responses? For example the response for `eth_getBalance`. Because when I do call the `fetchBalance` method, I can see that the same request is called again.

**SDK Version**
 - Version 2.0.0-rc.2

**To Reproduce**
1. Connect a wallet through web3modal
2. Watch the chrome dev tools "network" tab

**Expected behavior**
From my understanding, the calls for getting balance, or ENS related stuff should only be called when their methods are called.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome

### LINKED ISSUE #825: Prefix all web3modal related css variables with --w3m-
We are injecting web3modal's css variables into html tag of the page and current names for these might conflict with css vars defined within the user app itself, so we should prefix all our ones with `--w3m-` to avoid conflicts.

### LINKED ISSUE #826: Add more default chains (wagmi chains / images)
Go over wagmi supported chains / top charts for chains by mcap and defi TVL and add their images to default presets. i.e. Gnosis

### LINKED ISSUE #838: Handle disconnection triggered from wagmi
At the moment certain cleanup tasks are done when users click "disconnect" from modal's ui, however these same tasks need to be performed when user disconnects from external source i.e. wagmi / wallet.

We need to set up a listener for this (probably check for account being undefined in `watchAccount`)

### LINKED ISSUE #840: Document edge case when using localhost / ip tunnel for mobile testing
Several reports came in from users who were tunneling their dapp to mobile via localhost / ip setup. Some wallets either threw error or did nothing in this case (potentially a security feature?). Hence for mobile testing to work it needs to be deployed somewhere like vercel. Likely https over localhost / missing certs are also at play here.

I'm not sure if there is anything on our side to do here besides documenting this.

### LINKED ISSUE #846: All networks are shown as Ethereum
**Describe the bug**
After connecting metamask, any network is shown as Ethereum

**SDK Version**
 - 2.0.0

**To Reproduce**
1. Go to '...' https://docs.walletconnect.com/2.0/web3modal/react/installation
2. Follow steps to install and implement in react app
3.  Choose any network after connecting metamask
4. see that any network chosen does not change network icon, it stays ethereum

**Expected behavior**
After connecting to network, this network image should apear also 

const { selectedChain, setSelectedChain } = useWeb3ModalNetwork()

should show this network

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser chrome
 - Version 109.0.5414.76

**Additional context**

### PR DESCRIPTION
**Pull request by 0xAsimetriq**
## Breaking changes
- Make `projectId` required
- Removed `setSelectedChain`, `getSelectedChain` and `subscribeSelectedChain` in favor of using wagmi/core actions and new `setDefaultChain` action
- closes #846 (Removed `useWeb3ModalNetwork` hook in favor of using wagmi network hooks and `setDefaultChain` from `useWeb3Modal`)
- Update dependencies, new minimum requirements of `@wagmi/core` >=0.9 and `wagmi` >=0.11, see [wagmi migration guide](https://wagmi.sh/react/migration-guide)

## Changes
- closes #838 (Add missing state cleanup to watchAccount listener)
- closes #826 (Gnosis, EVMos, zkSync, Filecoin, Iotx, Metis, Moonbeam, Moonriver)
- closes #825 (prefix css variables)
- closes #824 (added `enableAccountView` config option that allows users who manage their own ui for accs to disable web3modal one thus not fetching balance / ens data etc unnecessarily)
- WalletConnect v2 support (WIP)
- (html) Added new `web3modal.setDefaultChain` action
- (react) Added `setDefaultChain` to `useWeb3Modal` hook
- closes #815 (memoise web3modal react component, there is no need for it to re-render unnecessarily)
- closes #840 (mobile localhost testing notes)
- Simplified react example
- Created `laboratory` example for consolidated testing

## Todo
- Mention required wagmi version for v2 support in the docs
- Update web3modal.com and add some of the new chains

### CODE REVIEW

**Comment by crypblizz8** [comment]
Feel free to ignore if still a WIP

For Custom switching, the ETH icon does not seem to appear.

https://user-images.githubusercontent.com/45455218/215447624-5ed2c87a-b4b8-42d8-bab2-ae39a37ef69d.mov

**Comment by 0xAsimetriq** [comment]
> Feel free to ignore if still a WIP
> 
> For Custom switching, the ETH icon does not seem to appear.
> 
>  Custom.mov

This one is expected 👍 custom will show placeholders if user didn't provide image (custom is essentially version without projectid. However we might be getting rid of it for v2)

File: packages/react/src/components/Web3Modal.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ interface Props extends Omit<ConfigCtrlState, 'enableStandaloneMode' | 'standalo
 /**
  * Component
  */
-export function Web3Modal({ ethereumClient, ...config }: Props) {
+function CreateWeb3Modal({ ethereumClient, ...config }: Props) {
```
Comment (crypblizz8) [inline packages/react/src/components/Web3Modal.tsx:17]: NB: Why was this renamed?

Comment (0xAsimetriq) [reply packages/react/src/components/Web3Modal.tsx:17]: to export it as default `Web3Modal` via memo here (names were clashing) https://github.com/WalletConnect/web3modal/pull/845/files/62d84f0a4816ac325ae1dac83391ee2c3120280a#diff-3477080de4f7c2140710399a4ede9295341ea1680f353b105b0558095da90132R34

**Review by crypblizz8 — DISMISSED** [review]
LGTM, awesome changes for the new chains / replacement of useWeb3ModalNetwork, V2 Config + more.

Could you point me to the RPC calls for the Balance / ENS part? I probably missed it.

~~Also curious on the createWeb3Modal name change as that may confuse previous W3MV2 developers?~~ Resolved

**Comment by 0xAsimetriq** [comment]
> LGTM, awesome changes for the new chains / replacement of useWeb3ModalNetwork, V2 Config + more.
> 
> Could you point me to the RPC calls for the Balance / ENS part? I probably missed it.
> 
> ~Also curious on the createWeb3Modal name change as that may confuse previous W3MV2 developers?~ Resolved

Sure thing, here is one for ens https://github.com/WalletConnect/web3modal/pull/845/files#diff-f86f8a7015a2a9167a5b113184e0e0686e27f3370f944fc93535d2c74eb44012R102 essentially these are now guarded by config flag and  not called when acc view is disabled.

**Pull request merged**